app/code/Magento/CatalogUrlRewrite/Model/ProductScopeRewriteGenerator.php

```diff
@@ -276,4 +283,83 @@ private function isCategoryRewritesEnabled()
     {
         return (bool)$this->config->getValue('catalog/seo/generate_category_product_rewrites');
     }
+
+    /**
+     * Return product changed categories store root category ids
```

Comment:
This description is pretty confusing. Could please change to something like...
```suggestion
     * Retrieve affected store ids for added/removed categories
```